test/e2e/e2e_suite_test.go

```diff
@@ -116,64 +114,41 @@ var _ = SynchronizedBeforeSuite(func() []byte {
 		}
 	}
 
+	bmoIronicNamespace := "baremetal-operator-system"
+
 	if e2eConfig.GetVariable("DEPLOY_IRONIC") != "false" {
 		// Install Ironic
 		By("Installing Ironic")
 		kustomization := e2eConfig.GetVariable("IRONIC_KUSTOMIZATION")
-		manifest, err := buildKustomizeManifest(kustomization)
-		Expect(err).NotTo(HaveOccurred())
-		err = clusterProxy.Apply(ctx, manifest)
+		err := BuildAndInstallDeployment(ctx, &BuildAndInstallDeploymentInput{
+			Kustomization:       kustomization,
```

Comment:
nit: we can skip the extra variable here I think.
```suggestion
			Kustomization:       e2eConfig.GetVariable("IRONIC_KUSTOMIZATION"),
```